File: FrontEnd/src/services/carePlanService.js
import api from "./api";

// =====================================================
// GET LATEST CARE PLAN BY PATIENT
// =====================================================

export const getLatestCarePlan = async (patientId) => {
    const response = await api.get(
        `/careplans/patient/${encodeURIComponent(patientId)}/latest`
    );

    return response.data;
};

// =====================================================
// GET CARE PLANS BY PATIENT
// =====================================================

export const getCarePlansByPatient = async (patientId) => {
    const response = await api.get(
        `/careplans/patient/${encodeURIComponent(patientId)}`
    );

    return response.data;
};

// =====================================================
// GET CARE PLAN BY ID
// =====================================================

export const getCarePlanById = async (id) => {
    const response = await api.get(`/careplans/${id}`);

    return response.data;
};

// =====================================================
// GET ALL CARE PLANS
// =====================================================

export const getAllCarePlans = async () => {
    const response = await api.get("/careplans");

    return response.data;
};

// =====================================================
// CREATE CARE PLAN
// =====================================================

export const createCarePlan = async (carePlan) => {
    const response = await api.post("/careplans", carePlan);

    return response.data;
};

// =====================================================
// UPDATE CARE PLAN
// =====================================================

export const updateCarePlan = async (id, carePlan) => {
    const response = await api.put(
        `/careplans/${id}`,
        carePlan
    );

    return response.data;
};

// =====================================================
// APPROVE CARE PLAN
// =====================================================

export const approveCarePlan = async (
    id,
    doctorId,
    doctorName,
    doctorNotes
) => {
    const params = new URLSearchParams();

    params.set("doctorId", doctorId || "");
    params.set("doctorName", doctorName || "Doctor");

    if (doctorNotes) {
        params.set("doctorNotes", doctorNotes);
    }

    const response = await api.put(
        `/careplans/${id}/approve?${params.toString()}`
    );

    return response.data;
};

// =====================================================
// REJECT CARE PLAN
// =====================================================

export const rejectCarePlan = async (
    id,
    doctorId,
    doctorName,
    doctorNotes
) => {
    const params = new URLSearchParams();

    params.set("doctorId", doctorId || "");
    params.set("doctorName", doctorName || "Doctor");
    params.set("doctorNotes", doctorNotes || "");

    const response = await api.put(
        `/careplans/${id}/reject?${params.toString()}`
    );

    return response.data;
};

// =====================================================
// DELETE CARE PLAN
// =====================================================

export const deleteCarePlan = async (id) => {
    await api.delete(`/careplans/${id}`);
};

File: CarePlanService/src/main/java/com/infosys/CarePlanService/service/CarePlanService.java
package com.infosys.CarePlanService.service;

import com.infosys.CarePlanService.model.CarePlan;

import java.util.List;

public interface CarePlanService {

    // =====================================================
    // AI CARE PLAN
    // =====================================================

    CarePlan generateAICarePlan(String patientId);

    // =====================================================
    // CREATE
    // =====================================================

    CarePlan createCarePlan(CarePlan carePlan);

    // =====================================================
    // GET
    // =====================================================

    List<CarePlan> getAllCarePlans();

    CarePlan getCarePlanById(String id);

    // =====================================================
    // PATIENT
    // =====================================================

    List<CarePlan> getCarePlansByPatientId(String patientId);

    CarePlan getLatestCarePlan(String patientId);

    // =====================================================
    // STATUS
    // =====================================================

    List<CarePlan> getCarePlansByStatus(String status);

    // =====================================================
    // DOCTOR
    // =====================================================

    List<CarePlan> getCarePlansByDoctorId(String doctorId);

    // =====================================================
    // RISK
    // =====================================================

    List<CarePlan> getCarePlansByRiskLevel(String riskLevel);

    // =====================================================
    // UPDATE
    // =====================================================

    CarePlan updateCarePlan(
            String id,
            CarePlan carePlan
    );

    // =====================================================
    // APPROVE
    // =====================================================

    CarePlan approveCarePlan(
            String id,
            String doctorId,
            String doctorName,
            String doctorNotes
    );

    // =====================================================
    // REJECT
    // =====================================================

    CarePlan rejectCarePlan(
            String id,
            String doctorId,
            String doctorName,
            String doctorNotes
    );

    // =====================================================
    // ADHERENCE
    // =====================================================

    CarePlan updateAdherence(
            String id,
            Integer adherencePercentage
    );

    // =====================================================
    // OUTCOME
    // =====================================================

    CarePlan updateOutcome(
            String id,
            Double currentRiskPercentage,
            String outcome
    );

    // =====================================================
    // DELETE
    // =====================================================

    void deleteCarePlan(String id);
}